### PR TITLE
Cargo - Make player carry/drag item on unload

### PR DESCRIPTION
**Pull request by Salluci**
**When merged this pull request will:**
- Title.

Player will start carrying or dragging item after unloading from cargo if possible. Includes weight checking.

### IMPORTANT

- [ ] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3mod.com/).
- [x] [Development Guidelines](https://ace3mod.com/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/cargo/functions/fnc_unloadCarryItem.sqf [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+if !(["ace_dragging"] call EFUNC(common,isModLoaded)) exitWith {};
+
+switch (true) do {
+    case ([_unloader, _object] call EFUNC(dragging,canCarry)): {
+        [_unloader, _object] call EFUNC(dragging,startCarry);
+    };
+    case ([_unloader, _object] call EFUNC(dragging,canDrag)): {
+        [_unloader, _object] call EFUNC(dragging,startDrag);
+    };
+};
```
Comment (jonpas) [inline addons/cargo/functions/fnc_unloadCarryItem.sqf:29]:
```suggestion
if ([_unloader, _object] call EFUNC(dragging,canCarry)) then {
    [_unloader, _object] call EFUNC(dragging,startCarry);
} else {
    if ([_unloader, _object] call EFUNC(dragging,canDrag)) then {
        [_unloader, _object] call EFUNC(dragging,startDrag);
    };
};
```
I find this way more readable.

Comment (Salluci) [reply addons/cargo/functions/fnc_unloadCarryItem.sqf:29]: Ew, nested if blocks.

As you wish.

**Review by PabstMirror — CHANGES_REQUESTED** [review]
doesn't work in mp

File: addons/cargo/functions/fnc_unloadCarryItem.sqf [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [player, object] call ace_cargo_fnc_unloadCarryItem
+ *
+ * Public: No
+ */
+params ["_unloader", "_object"];
```
Comment (PabstMirror) [inline addons/cargo/functions/fnc_unloadCarryItem.sqf:18]:
```
params ["_unloader", "_object"];
TRACE_2("unloadCarryItem-start",_unloader,_object);

if !(["ace_dragging"] call EFUNC(common,isModLoaded)) exitWith {};

// When unloading attached objects, this code will run before server has finished moving object to the safe position
[{
    params ["_unloader", "_object"];
    (_unloader distance _object) < 10
}, {
    params ["_unloader", "_object"];
    TRACE_2("unloadCarryItem-unloaded",_unloader,_object);
    if ([_unloader, _object] call EFUNC(dragging,canCarry)) exitWith {
        [_unloader, _object] call EFUNC(dragging,startCarry);
    };
    if ([_unloader, _object] call EFUNC(dragging,canDrag)) exitWith {
        [_unloader, _object] call EFUNC(dragging,startDrag);
    };
}, _this, 1.1, { TRACE_1("unloadCarryItem-failed to unload nearby player",_this); }] call CBA_fnc_waitUntilAndExecute;
```

When unloading a object type in mp the distance check in dragging was failing

`ace_cargo_fnc_unloadItem` calls `QGVAR(serverUnload)` and then immeditly calls `unloadCarryItem`
but it takes a bit for the server to actually move the object down to where the player is

this adds a little delay so all that can happen (only local mp it's 4 frames)
It's possible for this to fail if the unload position is away from the player, but that's just something that can happen and not really an error

also I was worried about collision and physx but it seemed to behave ok
I also switched the nested if statements to the superior `exitWith` style

Comment (Salluci) [reply addons/cargo/functions/fnc_unloadCarryItem.sqf:18]: Don't have access to anything that can (conveniently) commit atm, but works for me. Thanks for testing.

**Review by PabstMirror — APPROVED** [review]

**Pull request merged**